src/CodeGenerator/src/Generator/ResponseParser/RestJsonParser.php

```diff
@@ -308,20 +298,32 @@ private function parseResponseMap(MapShape $shape, string $input, bool $required
                         }
 
                         return $items;
-                    };';
+                    ';
                 }
 
-                $this->functions[$keyName] = strtr($body, [
-                    'FUNCTION_KEY' => \var_export($keyName, true),
+                $this->functions[$functionName] = $this->createPopulateMethod($functionName, strtr($body, [
                     'MAP_KEY' => var_export($locationName, true),
                     'MAP_ACCESSOR' => $this->parseElement(sprintf('$item[\'%s\']', $inputAccessorName), $shapeValue->getShape(), false),
-                ]);
+                ]));
             }
         }
 
-        return strtr($required ? '$fn[FUNCTION_KEY](INPUT)' : 'empty(INPUT) ? [] : $fn[FUNCTION_KEY](INPUT)', [
+        return strtr($required ? '$this->FUNCTION_NAME(INPUT)' : 'empty(INPUT) ? [] : $this->FUNCTION_NAME(INPUT)', [
             'INPUT' => $input,
-            'FUNCTION_KEY' => \var_export($keyName, true),
+            'FUNCTION_NAME' => $functionName,
         ]);
     }
+
+    private function createPopulateMethod(string $functionName, string $body): Method
+    {
+        $method = new Method($functionName);
+        $method->setVisibility(ClassType::VISIBILITY_PRIVATE)
```

Comment:
Have a look at #677. Sometimes it is easy to find out the return type. We should add that as a comment.

Reply:
good idea! it also removes a lot of ignored rule in psalm